Marlin/src/inc/SanityCheck.h

```diff
@@ -822,15 +822,17 @@ static_assert(Y_MAX_LENGTH >= Y_BED_SIZE, "Movement bounds (Y_MIN_POS, Y_MAX_POS
  * Granular software endstops (Marlin >= 1.1.7)
  */
 #if ENABLED(MIN_SOFTWARE_ENDSTOPS) && DISABLED(MIN_SOFTWARE_ENDSTOP_Z)
-  #if IS_KINEMATIC
+  #if ENABLED(POLARGRAPH)
+  #elif IS_KINEMATIC
```

Comment:
I think you're missing an `#error` here. There's nothing in this `#if`, so it's simply skipped.

Reply:
there should be nothing.  the IS_KINEMATIC error is intended for deltas, not polargraph.

Reply:
Then the check should be added to the parent `#if`:

`#if ENABLED(MIN_SOFTWARE_ENDSTOPS) && DISABLED(MIN_SOFTWARE_ENDSTOP_Z) && DISABLED(POLARGRAPH)`